docs(weave): Update platform doc with dedicated cloud info (#3244)
wandb/weave: docs/docs/guides/platform/index.md

- **Role-based access control:** Configure access at the [team](https://docs.wandb.ai/guides/hosting/iam/manage-organization#assign-or-update-a-team-members-role) or [project](https://docs.wandb.ai/guides/hosting/iam/restricted-projects#project-level-roles) level to ensure users access data on a need-to-know basis.
- **Scoped service accounts:** Automate Gen AI workflows using service accounts scoped to your organization or team.
- **[SCIM API and Python SDK](https://docs.wandb.ai/guides/hosting/iam/automate_iam):** Manage users and teams efficiently with SCIM API and Python SDK.

## Data Security

- **SaaS Cloud:** Data for all Weave users is stored in a shared Clickhouse Cloud cluster, encrypted using cloud-native encryption. Shared compute services process the data, ensuring isolation through a security context comprising your W&B organization, team, and project.

- **Dedicated Cloud:** Data is stored in a unique Clickhouse Cloud cluster in the cloud and region of your choice. A unique compute environment processes the data, with the following additional protections:

>   - **IP allowlisting:** Authorize access to your instance from specific IP addresses.

Reviewer: ```suggestion
  - **[IP allowlisting](https://docs.wandb.ai/guides/hosting/data-security/ip-allowlisting):** Authorize access to your instance from specific IP addresses. This is an optional capability.
```

>   - **Private connectivity:** Route data securely through the cloud provider's private network.

Reviewer: ```suggestion
  - **[Private connectivity](https://docs.wandb.ai/guides/hosting/data-security/private-connectivity):** Route data securely through the cloud provider's private network. This is an optional capability.
```

>   - **Data encryption:** Encrypt data at rest using a unique W&B-managed encryption key.

Reviewer: ```suggestion
  - **[Data encryption](https://docs.wandb.ai/guides/hosting/data-security/data-encryption):** W&B encrypts data at rest using a unique W&B-managed encryption key.
```

>   - **Strong encryption mechanisms:** File-based encryption using Clickhouse Cloud's capabilities.

Reviewer: ```suggestion
  - **Clickhouse cluster security:** W&B connects to the unique Clickhouse Cloud cluster for your Dedicated Cloud instance over the cloud provider's private network. W&B also encrypts the cluster using a unique W&B-managed encryption key, while leveraging Clickhouse's file level encryption.
```

:::important

> [The W&B Platform secure storage connector](https://docs.wandb.ai/guides/hosting/data-security/secure-storage-connector/?_gl=1*jzcfye*_gcl_au*OTI3ODM1OTcyLjE3MzE0MzU1NjUuMTYxOTQ4Mzk1LjE3MzMyNTYwMTYuMTczMzI1NjAxNQ..*_ga*ODEyMjQ4MjkyLjE3MzE0MzU1NjU.*_ga_JH1SJHJQXJ*MTczNDEyNjAxNy45Ny4xLjE3MzQxMjg5NDEuNjAuMC4w*_ga_GMYDGNGKDT*MTczNDEyNzgzMy44MS4xLjE3MzQxMjg1MTAuMC4wLjA.) is not available in Weave.

Reviewer: ```suggestion
[The W&B Platform secure storage connector or BYOB](https://docs.wandb.ai/guides/hosting/data-security/secure-storage-connector) is not available for Weave.
```
